lib/inspec/rspec_json_formatter.rb

```diff
@@ -586,43 +585,54 @@ def print_anonymous_control(control)
 
   def print_profile_summary
     summary = profile_summary
+    return unless summary['total'] > 0
 
-    s = format('Profile Summary: %s%d successful%s, %s%d failures%s, %s%d skipped%s',
-               COLORS['passed'], summary['passed'], COLORS['reset'],
-               COLORS['failed'], summary['failed']['total'], COLORS['reset'],
-               COLORS['skipped'], summary['skipped'], COLORS['reset'])
+    s = format('Profile Summary: %s, %s, %s',
+               format_with_color('passed', "#{summary['passed']} successful"),
+               format_with_color('failed', "#{summary['failed']['total']} failures"),
+               format_with_color('skipped', "#{summary['skipped']} skipped"),
+              )
     output.puts(s) if summary['total'] > 0
   end
 
   def print_tests_summary
     summary = tests_summary
 
-    s = format('Test Summary: %s%d successful%s, %s%d failures%s, %s%d skipped%s',
-               COLORS['passed'], summary['passed'], COLORS['reset'],
-               COLORS['failed'], summary['failed'], COLORS['reset'],
-               COLORS['skipped'], summary['skipped'], COLORS['reset'])
+    s = format('Test Summary: %s, %s, %s',
+               format_with_color('passed', "#{summary['passed']} successful"),
+               format_with_color('failed', "#{summary['failed']} failures"),
+               format_with_color('skipped', "#{summary['skipped']} skipped"),
+              )
+
     output.puts(s)
   end
 
   # Formats the line (called from print_line)
   def format_line(fields)
-    format = '%color%indicator%id%summary'
+    format = '%indicator%id%summary'
     format.gsub(/%\w+/) do |x|
       term = x[1..-1]
       fields.key?(term.to_sym) ? fields[term.to_sym].to_s : x
-    end + COLORS['reset']
+    end
   end
 
   # Prints line; used to print results
   def print_line(fields)
-    output.puts(format_line(fields))
+    output.puts(format_with_color(fields[:color], format_line(fields)))
   end
 
   # Helps formatting summary lines (called from within print_line arguments)
   def format_lines(lines, indentation)
     lines.gsub(/\n/, "\n" + indentation)
   end
 
+  def format_with_color(color_name, text)
+    return text unless RSpec.configuration.color
```

Comment:
Okay, makes sense. Its time to make reporting independent or RSpec. Nothing we can solve in that PR though.